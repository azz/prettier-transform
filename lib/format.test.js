"use strict";

const format = require("./format");

const replaceFooWithBar = require("../test/foo-bar");
const replaceBarWithBaz = require("../test/bar-baz");

describe("format()", () => {
  test("throws an error if code is not a string", () => {
    expect(() => {
      format(null, [], { parser: "babylon" });
    }).toThrowErrorMatchingSnapshot();
  });

  test("throws an error if transforms is not an array of functions", () => {
    expect(() => {
      format("foo()", null, { parser: "babylon" });
    }).toThrowErrorMatchingSnapshot();
  });

  test("throws an error if no parser option is present", () => {
    expect(() => {
      format("foo()", [], null);
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      format("foo()", [], {});
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      format("foo()", [], { parser: null });
    }).toThrowErrorMatchingSnapshot();
  });

  it("outputs code formatted by prettier", () => {
    const output = format("foo()", [], { parser: "babylon" });
    expect(output).toEqual("foo();\n");
  });

  it("applies multiple transforms in order", () => {
    const output = format("foo()", [replaceFooWithBar, replaceBarWithBaz], {
      parser: "babylon"
    });
    expect(output).toEqual("baz();\n");
  });

  it("forwards options to prettier", () => {
    const output = format("foo()", [], {
      parser: "babylon",
      semi: false
    });
    expect(output).toEqual("foo()\n");
  });
});
