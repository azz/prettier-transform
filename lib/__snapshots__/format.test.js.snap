// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format() throws an error if code is not a string 1`] = `"text must be a string"`;

exports[`format() throws an error if no parser option is present 1`] = `"prettierOptions must be an object"`;

exports[`format() throws an error if no parser option is present 2`] = `"parser must be a provided"`;

exports[`format() throws an error if no parser option is present 3`] = `"parser must be a provided"`;

exports[`format() throws an error if transforms is not an array of functions 1`] = `"transforms must be an array of functions"`;
